### PR TITLE
Add digital twin lifecycle sample

### PR DESCRIPTION
**Pull request by abhipsaMisra**
Add code for sample cleanup and creation of digital twin.

TODO: 
- Snippets
- readme

### CODE REVIEW

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleSample.java [hunk 1]
```diff
@@ -72,21 +97,71 @@ public static void deleteTwins() throws IOException, InterruptedException {
...
-            .forEach((twinId, twinContent) -> client.deleteDigitalTwin(twinId)
-                .doOnSuccess(aVoid -> System.out.println("Deleted digital twin: " + twinId))
-                .doOnError(throwable -> {
-                    // If digital twin does not exist, ignore.
-                    if (!(throwable instanceof ErrorResponseException) || !((ErrorResponseException) throwable).getValue().getError().getCode().equals("DigitalTwinNotFound")) {
-                        System.err.println("Could not delete digital twin " + twinId + " due to " + throwable);
+            .forEach((twinId, twinContent) -> {
+                // Call APIs to delete all relationships.
+                client.listRelationships(twinId, BasicRelationship.class)
+                    .doOnComplete(deleteRelationshipsSemaphore::release)
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleSample.java:107]: This runnable is executed when the flux completes successfully.

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleSample.java [hunk 2]
```diff
@@ -6,7 +6,9 @@
 import com.azure.core.http.policy.HttpLogDetailLevel;
```
Comment (azabbasi) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleSample.java:6]: I think we should have an Async Suffix in the file name to indicate that we are using Async APIs for this sample.

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleSample.java [hunk 3]
```diff
@@ -72,21 +97,71 @@ public static void deleteTwins() throws IOException, InterruptedException {
...
-                .doOnSuccess(aVoid -> System.out.println("Deleted digital twin: " + twinId))
-                .doOnError(throwable -> {
-                    // If digital twin does not exist, ignore.
-                    if (!(throwable instanceof ErrorResponseException) || !((ErrorResponseException) throwable).getValue().getError().getCode().equals("DigitalTwinNotFound")) {
-                        System.err.println("Could not delete digital twin " + twinId + " due to " + throwable);
+            .forEach((twinId, twinContent) -> {
+                // Call APIs to delete all relationships.
+                client.listRelationships(twinId, BasicRelationship.class)
+                    .doOnComplete(deleteRelationshipsSemaphore::release)
+                    .doOnError(throwable -> {
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleSample.java:108]: This function is executed when the flux completes with an error

**Review by azabbasi — APPROVED** [review]

File: sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+    /**
+     * Creates all twins specified in the DTDL->DigitalTwins directory.
+     * @throws IOException If an I/O error is thrown when accessing the starting file.
+     * @throws InterruptedException If the current thread is interrupted while waiting to acquire permits on a semaphore.
+     */
+    public static void createTwins() throws IOException, InterruptedException {
+        System.out.println("CREATE DIGITAL TWINS");
+        Map<String, String> twins = FileHelper.loadAllFilesInPath(TwinsPath);
+        final Semaphore createTwinsSemaphore = new Semaphore(0);
```
Comment (bikamani) [inline sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java:181]: To understand why do we need semaphore here? Do we have some limitations for parallel operations?

Comment (abhipsaMisra) [reply sdk/digitaltwins/azure-digitaltwins-core/src/samples/java/com/azure/digitaltwins/core/DigitalTwinsLifecycleAsyncSample.java:181]: No, not really. The semaphore is to ensure that we do not exit before the async operation has completed. We start all async operations and then release the semaphore only once the async call has completed.
That way, the subsequent operations are executed after the previous call has completed -> similar to doing a await on a Task.

**Pull request merged**